### DISCUSSION TITLE
Upgrade from pdns-auth 4.8 to 4.9

### DISCUSSION THREAD

**Discussion by git-noise** [PowerDNS/pdns#14814] [Q&A] [answered]
Hello,

I am filling this here, as its probably as much support than bug.

After updating from 4.8 to 4.9, I am getting an error `"while filling the zone cache"`.

```
PDNSException while filling the zone cache: Database error trying to retrieve all domains:Could not prepare statement: select domains.id, domains.name, records.content, domains.type, domains.master, domains.notified_serial, domains.last_check, domains.account, domains.catalog from domains LEFT JOIN records ON records.domain_id=domains.id AND records.type='SOA' AND records.name=domains.name WHERE records.disabled=0 OR ?: Unknown column 'domains.catalog' in 'field list'
```

I suspect this is linked to the following - https://doc.powerdns.com/authoritative/upgrading.html#gsql-backends
```
Also, get-all-domains-query got an extra column for a zone’s catalog assignment.
```

Is it possible that the instructions for upgrading are missing a SQL query to add the column, or that the schema needs to be updated - I only found the 4.7 schema online?

Best,

**Comment by Habbie** [comment, marked as answer]
The 4.7 schema has the catalog column; however, it is possible you never noticed it missing because that query did not look for it. In other words, it looks like you may have failed to apply the 4.7 update in the past.

**Comment by git-noise** [comment]
Indeed that was part of the issue, some of my backends were missing the schema changes. Thank you and sorry for the noise!
